addon/components/column-chart/component.js

```diff
@@ -462,17 +498,27 @@ export default BaseChartComponent.extend({
 
         // Choose the tallest bar in the stack (lowest y value) and place the max/min labels above that.
         // Avoids label falling under any bar in the stack.
-        let yValues = [];
+        let yValues = { y: [], height: [] };
         this.get('chart').selectAll('.sub rect.bar').each(function () {
-            yValues.push(parseInt(d3.select(this).attr('y')));
+            yValues.y.push(parseInt(d3.select(this).attr('y')));
+            yValues.height.push(parseInt(d3.select(this).attr('height')));
         });
-        const maxLabelY = Math.min(...yValues);
-
+        const maxLabelY = Math.min(...yValues.y);
+        const maxLabelYHeight = Math.max(...yValues.height);
+
+        // if label collision  detection is turned on, then raise the max and min labels slightly above data values.
+        let raiseLabelBy = 0;
+        if (this.get('labelOptions.labelCollisionResolution') === 'auto') {
+            raiseLabelBy = 10;
```

Comment:
Would it be possible to do collision detection (via bounding client rectangle) to determine whether to adjust the y of the min/max? That way min/max won't change position unless another label before it is causing problems. I think you would just need to iterate across the bars again to determine collision with the max/min

Reply:
Instead of raising the max/min labels above the other data values can we just not show any data labels that would collide with the max/min? It looks a lot nicer when all of the labels are on the same level

Reply:
We can, it's just that the collision detection runs inside the addDataValues() and the addMaxMin() is a separate function, so we have to implement some kind of signal/indication that max/min has a nearby collision.

choices - 
Either do another collision detection logic inside the addMaxMin(), or do max/min stuff inside the addDataValues() instead of addMaxMin().

or probably in the next iteration?

Reply:
It does look better when we display on one line. Could we accept that it will look strange for now (but not collide) and make a followup change to keep everything on one line?

Implementation-wise, I think the chart would just have to place the min/max value first. Then as it iterates through values it would first check to see if it collides with the min/max. If it does, skip to the next item. If it doesn't, apply the skip logic outlined above. @KalaRaja

Reply:
agreed, raising the labels looks a bit off.

Reply:
@shanebruggeman we can fix it separately if you really think we need to get this in. My preference would be to do it right the first time but I understand if that causes problems.

Reply:
I'm gonna take a stab at it.

Reply:
The min and max labels are now shown in one line, the data labels that collide with min and max are removed by the logic explained in the description.

Reply:
awesome. thanks!

Reply:
reverted the changes, writing new and better logic.

Reply:
guys, please read the description and review the new changes, thanks